src/workers/python/papyros/papyros.py

```diff
@@ -124,6 +124,17 @@ async def run_async(self, source_code, mode="exec", top_level_await=True):
                     while isinstance(result, Awaitable):
                         result = await result
                     return result
+            except ModuleNotFoundError as mnf:
+                # Try to automatically install missing dependencies
+                # As they sometimes might be hidden within libraries
+                try:
+                    self.callback("loading", data=dict(loading=True, modules=[mnf.name]), contentType="application/json")
+                    await self.install_imports(f"import {mnf.name}", ignore_missing=False)
+                    self.callback("loading", data=dict(loading=False, modules=[mnf.name]), contentType="application/json")
+                    return await self.run_async(source_code, mode=mode, top_level_await=top_level_await)
+                except:
+                    # If the module is truly not findable, raise the error again
+                    raise mnf
```

Comment:
This will translate regular exceptions in the user's code into ModuleNotFoundError even if the import installed just fine.